test-data/unit/check-async-await.test

```diff
@@ -1013,13 +1013,17 @@ async def foo(x: int) -> int: ...
 
 # These are allowed in some cases:
 top_level = await foo(1)  # E: "await" outside function  [top-level-await]
-crasher = [await foo(x) for x in [1, 2, 3]]  # E: "await" outside function  [top-level-await]
+crasher = [await foo(x) for x in [1, 2, 3]]  # E: "await" outside coroutine ("async def")  [await-not-async] \
```

Comment:
More explicit tests would be really helpful here. Consider at least checking all of the following in different scopes (module level, non-async function body and async function body at least; the former two should produce the same errors and async function should allow all of those):

```python
(await x for x in xs)  # OK
[await x for x in xs]  # E
{await x for x in xs}  # E
{0: await x for x in xs}  # E
{await x: 0 for x in xs}  # E

(x async for x in xs)  # OK
[x async for x in xs]  # E
{x async for x in xs}  # E
{x: 0 async for x in xs}  # E

(x for x in await xs)  # E
[x for x in await xs]  # E
{x for x in await xs}  # E
{x: 0 for x in await xs}  # E
```

There's already `testAsyncForOutsideCoroutine` for the 2nd block, the 3rd one is probably not interesting (follows regular rules), but the 1st one isn't exercised yet.